src/simple_tracker_observer/simple_tracker_observer/cloud_estimator.py

```diff
@@ -0,0 +1,180 @@
+# Original work Copyright (c) 2022 Sky360
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+import cv2
+import os
+import sys
+import numpy as np
+import math
+
+class CloudEstimator():
```

Comment:
Is this your original algorithm?

Reply:
No David wrote, I am just integrating it into the tracker.

Reply:
Thanks guys. I pulled together some algorithms from existing research - they are used fairly commonly across the literature and are based on simple thresholding.  Works well for our purposes though I think! I provide a breakdown with references here: https://github.com/Sky360-Repository/cloud_estimation/blob/main/cloud_estimation.ipynb